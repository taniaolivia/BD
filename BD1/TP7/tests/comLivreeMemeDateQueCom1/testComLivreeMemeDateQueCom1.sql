-- ============================================================================
-- Fichier          : testComLivreeMemeDateQueCom1.sql
-- Auteur           : Tania OLIVIA
-- Date             : 6 Decembre 2019
-- Base				: Surgeles
-- Role             : Le test pour afficher les commandes livrées à la même date que la commande de code 1
-- Fichier resultat : testComLivreeMemeDateQueCom1.out
-- ============================================================================

-- ============================================================================
-- Modification du format de la date pour se conformer aux donnees du programme de test
-- ============================================================================
alter session set NLS_DATE_FORMAT='dd/mm/yyyy';

-- ============================================================================
-- Suppression des nuplets de la base de donnees
-- NB : Le test est effectué sur une base vide, il est nécessaire 
--      de supprimer tous les nuplets de toutes les relations
-- ============================================================================
@ ../../utilitaires/vider_base_surgeles.sql
-- ============================================================================
-- Jeu de test pour le DOMAINE VALIDE 
-- NB : L'insertion de chaque nuplet doit faire l'objet d'un commentaire, 
-- notamment pour motiver le/les cas dans le(s)quel(s) il intervient.
-- ============================================================================

-- CAS1
-- Un client qui a passé une commande
------- Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(1, 'MARTIN', 'Jean', '0605050505', 'martin@example.org', 'La Rochelle'); 
------- La commande passée par le client
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(1, 1, '01/10/2013', '20/10/2013');
------- Le liveur de la tournée
INSERT INTO LIVREUR (IDLIV, NOMLIV)
VALUES(1, 'BERTRAND');
------- La livraison
INSERT INTO LIVRAISON (IDLIV, IDCOM, DATELIV)
VALUES(1, 1, '25/11/2013');

-- CAS2
-- Un client qui a passé une commande et qui a le même date que la commande de code 1
------- Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(2, 'DUPONT', 'Jean', '0606060606', 'dupont@example.org', 'La Rochelle'); 
------- La commande passée par le client
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(2, 2, '01/10/2013', '25/10/2013');
------- La livraison
INSERT INTO LIVRAISON (IDLIV, IDCOM, DATELIV)
VALUES(1, 2, '25/11/2013');

-- CAS3
-- Un client qui a passé une commande et qui a le même date que la commande de code 1
------ Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(5, 'CAVIN', 'Pauline', '0608080808', 'cavin@example.org', 'La Rochelle'); 
------- La commande passée par le client
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(5, 5, '10/10/2013', '25/10/2013');
------- Le liveur de la tournée
INSERT INTO LIVREUR (IDLIV, NOMLIV)
VALUES(2, 'MAURICE');
------- La livraison
INSERT INTO LIVRAISON (IDLIV, IDCOM, DATELIV)
VALUES(2, 5, '25/11/2013');

-- ============================================================================
-- Jeu de test pour le DOMAINE INVALIDE 
-- ============================================================================

-- CAS4
-- Un client qui n'a pas passé une commande parce qu'il n'a pas le même date que la commande de code 1
------- Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(4, 'LALLIC', 'Tituan', '0707070707', 'lallic@example.org', 'La Rochelle'); 
------- La commande passée par le client
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(4, 4, '01/10/2013', '20/11/2013');
------- La livraison
INSERT INTO LIVRAISON (IDLIV, IDCOM, DATELIV)
VALUES(1, 4, '24/11/2013');

-- CAS5
-- Un client qui n'a pas passé une commande parce qu'il n'a pas le même date que la commande de code 1
------ Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(6, 'VOTIN', 'Elodie', '0609080808', 'votin@example.org', 'La Rochelle'); 
------- La commande passée par le client
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(6, 6, '05/10/2013', '25/10/2013');
------- La livraison
INSERT INTO LIVRAISON (IDLIV, IDCOM, DATELIV)
VALUES(2, 6, '26/11/2013');

-- ============================================================================
-- resultats a mettre dans le fichier a nommer obligatoirement:
-- 
-- ce fichier doit contenir:	
--	1. l'etat de la base avant execution de la requete (donc juste apres 
--         les INSERT)
--	2. le texte SQL de la requete
--	3. le resultat de l'execution de la requete
-- ============================================================================

-- Debut de l'ecriture du fichier resultat du programme de test 
SPOOL testComLivreeMemeDateQueCom1.out
PROMPT fichier resultat du test : testComLivreeMemeDateQueCom1.out

-- Appel du fichier de presentation
@ ../../utilitaires/presentation_surgeles.sql

-- Afficher le contenu des relations de la base de donnees
PROMPT Contenu de la base :
@ ../../utilitaires/afficher_base_surgeles.sql

-- ============================================================================
-- Requete a tester 
-- ============================================================================
prompt 'Resultat de la requete'
SET echo ON
@comLivreeMemeDateQueCom1.sql
SET echo OFF

-- Fin de l'ecriture du fichier resultat du programme de test 
SPOOL OFF

-- ============================================================================
-- Fin du programme de test
-- ============================================================================

