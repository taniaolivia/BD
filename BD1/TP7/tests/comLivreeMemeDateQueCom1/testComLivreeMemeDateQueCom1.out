fichier resultat du test : testComLivreeMemeDateQueCom1.out
Contenu de la base :
Contenu de la table Client :
=================================

     IDCLI NOMCLI     PRENOMCLI      TELCLI MAILCLI            ADRCLI                                                                                 
---------- ---------- ---------- ---------- ------------------ ------------                                                                           
         1 MARTIN     Jean       ########## martin@example.org La Rochelle                                                                            
         2 DUPONT     Jean       ########## dupont@example.org La Rochelle                                                                            
         5 CAVIN      Pauline    ########## cavin@example.org  La Rochelle                                                                            
         4 LALLIC     Tituan     ########## lallic@example.org La Rochelle                                                                            
         6 VOTIN      Elodie     ########## votin@example.org  La Rochelle                                                                            

Contenu de la table Commande:
=================================

     IDCOM      IDCLI DATECOM    DATERECEPP                                                                                                           
---------- ---------- ---------- ----------                                                                                                           
         1          1 01/10/2013 20/10/2013                                                                                                           
         2          2 01/10/2013 25/10/2013                                                                                                           
         5          5 10/10/2013 25/10/2013                                                                                                           
         4          4 01/10/2013 20/11/2013                                                                                                           
         6          6 05/10/2013 25/10/2013                                                                                                           

Contenu de la table Ligne_Commande:
=================================

aucune ligne sélectionnée

Contenu de la table Produit:
=================================

aucune ligne sélectionnée

Contenu de la table Livraison:
=================================

     IDLIV      IDCOM DATELIV                                                                                                                         
---------- ---------- ----------                                                                                                                      
         1          1 25/11/2013                                                                                                                      
         1          2 25/11/2013                                                                                                                      
         2          5 25/11/2013                                                                                                                      
         1          4 24/11/2013                                                                                                                      
         2          6 26/11/2013                                                                                                                      

Contenu de la table Livreur:
=================================

     IDLIV NOMLIV                                                                                                                                     
---------- ----------                                                                                                                                 
         1 BERTRAND                                                                                                                                   
         2 MAURICE                                                                                                                                    

'Resultat de la requete'
SQL> @comLivreeMemeDateQueCom1.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : comLivreeMemeDateQueCom1.sql
SQL> -- Auteur		 : Tania OLIVIA
SQL> -- Date		 : 6 Decembre 2019
SQL> -- Base			     : Surgeles
SQL> -- Role		 : Quelles sont les commandes livrées à la même date que la commande de code 1 ?
SQL> -- ============================================================================
SQL> 
SQL> SELECT C.IDCLI, D.IDCOM
  2  FROM CLIENT C, COMMANDE D
  3  WHERE C.IDCLI = D.IDCILI (SELECT L.DATELIV, I.IDCOM, V.IDLIV
  4  			      FROM COMMANDE I, LIVRAISON L, LIVREUR V
  5  			      WHERE I.IDCOM = 1
  6  			      AND I.IDCOM = L.IDCOM
  7  			      AND V.IDLIV = L.IDLIV
  8  			      AND D.IDCOM = I.IDCOM
  9  );
WHERE C.IDCLI = D.IDCILI (SELECT L.DATELIV, I.IDCOM, V.IDLIV
                          *
ERREUR à la ligne 3 :
ORA-00936: expression absente 


SQL> SET echo OFF
